### ISSUE TITLE
There are serious issues when used with the `react/self-closing-comp` / `react/jsx-sort-props` rules.

### ISSUE THREAD

**Issue by u3u** [mdx-js/eslint-mdx#488]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.2.0

### Steps to reproduce

```mdx
❌ Empty components are self-closing eslint(react/self-closing-comp)
<span>test</span>

💡 Fix this react/jsx-sort-props problem
<Image alt="alt" src="src" width="315" height="100" format="svg" />
↓↓↓↓↓
<Image alt="alt" format="src" height="315" src="100" width="svg" />
```

Clone this repository [u3u/eslint-config](https://github.com/u3u/eslint-config), and input the above content in any mdx file to reproduce.

### Expected behavior

Keep consistent with `mdx/code-blocks`.

### Actual behavior

1. `<span>test</span>` Regarded as an empty tag self-closing, it will be fixed to `<span />`, directly losing the content.
2. After automatic sorting of JSX attributes, the content and attributes no longer correspond.

### Runtime

Node v20

### Package manager

pnpm

### OS

macOS

### Build and bundle tools

esbuild

**Comment by JounQin** [comment]
OK, the `react/self-closing-comp` part is just as same as #437.

**Comment by u3u** [comment]
@JounQin Yes, they have no issues in the code block.

**Comment by JounQin** [comment]
@u3u Please help to test:

```sh
# yarn 1
yarn add https://pkg.csb.dev/mdx-js/eslint-mdx/commit/2dbcb20a/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/2dbcb20a/eslint-plugin-mdx
# yarn 2, 3
yarn add eslint-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/2dbcb20a/eslint-mdx/_pkg.tgz eslint-plugin-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/2dbcb20a/eslint-plugin-mdx/_pkg.tgz
# npm
npm i https://pkg.csb.dev/mdx-js/eslint-mdx/commit/2dbcb20a/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/2dbcb20a/eslint-plugin-mdx
```

**Comment by u3u** [comment]
🥳 Perfect!

https://github.com/mdx-js/eslint-mdx/assets/20062482/0d2a8073-8e0f-4280-a19c-6c1b14b909a1

**Comment by JounQin** [comment]
v2.2.1 has already been released.

**Comment by u3u** [comment]
Thanks!
